libs/sdk-ui-charts/prepublish.sh

```diff
@@ -0,0 +1,8 @@
+#! /bin/bash
+fswatch -r src/ | (while read;
+do
+ echo "Running prepublish.."
+ npm run prepublishOnly
+ cp -r ./dist ~/gd/gdc-analytical-designer/node_modules/\@gooddata/sdk-ui-charts
+ echo "Done!\n"
+done)
```

Comment:
please read docs/sdk-dev.md ; if you want to test stuff in AD/KD, you can use `yarn link`. combined with `dev` compilation mode this addresses what this script does.

note though, that when linking sdk-ui-charts, you need to also link the sdk-ui-ext (plug viz impl which also depends on the charts).

Reply:
additionally, I suggest to use the storybook in `sdk-ui-tests` for dev testing. that way you are confined to SDK-only for starters. when dev-testing like this within the monorepo, all the dependent packages are fully symlinked. so you all you need to do is run 'dev' compilation mode on charts

Reply:
Sorry the prepublish was committed by mistake. Thanks for the hint, though! The script pretty much works, but your solution surely seems like much better, I'll try it.

Reply:
When it comes to testing in storybooks, the feature I'm implementing is actually for AD, even though most of the code is in SDK.

Reply:
- you should test the configuration panel in storybook (see existing tests for inspiration).
- you should test chart configuration & updated chart behavior (react component) in storybook
- the testing infrastructure will generate insights (fka vis objects)
- stories will be automatically created to render insights using plug viz; this rendered insight will include all configuration panels with their setting + the actual render of the chart using plug viz

==> you can test probably 80+% in SDK and then do final tuning in AD.

if you have not done so, I encourage to check out sdk-ui-tests, its README and start the storybook to see what kind of stories we have.

with v8 we are aiming for greater encapsulation of SDK & its testing. ideally you should be able to verify everything in SDK and then 'just' do final verification in AD; this is not the case yet but you should be able to get quite far. besides, this particular feature probably needs 0 changes in AD codebase itself right?

Reply:
@lupko Yes, I didn't need to change anything in AD. Ok, I'll check the stories, I didn't know there were stories for configuration panels.